pages/models/generations/Sources/epri_gfm/index.md

```diff
@@ -0,0 +1,276 @@
+
+The model is a positive-sequence RMS model, hence it assumes symmetrical
+operating conditions and neglects high-frequency dynamics. This type of
+model is often used in large-scale stability studies, for which it
+reflects the relevant phenomena. It is not a detailed physical model of
+the unit. Also for some stability phenomena (e.g. resonance stability)
+this model is not sufficient and EMT models or other approaches may be
+necessary. A comparision against EMT is shown in the EPRI documentation
+. It is also available for
+[PSCAD](https://www.pscad.com/knowledge-base/download/PSCAD-EPRI-Generic-GFM.pdf&ved=2ahUKEwjOpr_z0rOKAxVPg_0HHUO_JAgQFnoECBoQAQ&usg=AOvVaw1LVPVn2jXqTdzbFxeCoPo7).
+
+## Model description
+
+### Difference between grid following and grid forming inverter
+
+A grid-following inverter synchronizes itself with the voltage and
+frequency values of the power grid using a Phase-Locked Loop (PLL). The
+PLL tracks the phase angle of the grid and therefore enables the
+inverter control to follow the grid. In contrast, a grid-forming
+inverter actively controls the grid’s voltage and frequency, enabling it
+to create or stabilize the grid, especially in scenarios where
+traditional grid infrastructure is weak or absent. Grid-forming
+inverters do not rely on a PLL. Many approaches synchronize with the
+grid through the power like a synchronous generator.
+
+### EPRI GFM operation modes
+
+The Generic EPRI GFM model offers four control modes:
+
+- Droop based, $$\omega_\mathrm{flag}=1$$, see
+  <a href="#fig-droop_mode" class="quarto-xref">Figure 1</a>
+- Virtual Synchronous Machine (VSM), $$\omega_\mathrm{flag}=2$$, see
+  <a href="#fig-vsm_mode" class="quarto-xref">Figure 2</a>
+- Dispatchable Virtual Oscillator (dVOC) based GFM,
+  $$\omega_\mathrm{flag}=3$$, see
+  <a href="#fig-dvoc_mode" class="quarto-xref">Figure 3</a>
+- Phase Locked Loop (Grid following mode), $$\omega_\mathrm{flag}=0$$, see
+  <a href="#fig-pll_mode" class="quarto-xref">Figure 4</a>
+
+## Model schema
+
+The different control method diagrams of the GFM is shown below:
+
+
+![](./drawings/Droop_Mode.drawio.svg)
+
+
+<div id="fig-droop_mode">
+Figure 1: Droop mode diagram
+</div>
+
+
+![](./drawings/VSM_Mode.drawio.svg)
+
+
+<div id="fig-vsm_mode">
+Figure 2: VSM mode diagram
+
+</div>
+
+![](./drawings/DVOC_Mode.drawio.svg)
+
+
+<div id="fig-dvoc_mode">
+Figure 3: dVOC mode diagram
+
+</div>
+
+![](./drawings/PLL_Mode.drawio.svg)
+
+
+<div id="fig-pll_mode">
+Figure 4: PLL mode diagram
+
+</div>
+
+> [!NOTE]
+>
+> The xy reference frame is the real – imaginary coordinate frame of the
+> network while the dq reference frame is the coordinate frame of the
+> control.The relative angle between these reference frames is denoted
+> by the control variable $$\theta_\mathrm{inv}$$.
+
+### D and q reference currents calculation
+
+
+![](./drawings/Grid_Forming.drawio.svg)
+
+
+<div id="fig-IdIqRefCalculation">
+Figure 5: Calculation of d and q reference currents
```

Comment:
for $$\omega_{flag} = {1,2,3}$$ you have two PI controllers controlling v_inv_d to different values. The upper one (I_d^ref) to v_ind_d^ref and the other (I_q^ref) to 0. I think the lower one should be v_inv_q.